src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java

```diff
@@ -2223,15 +2238,14 @@ public EntryResult tMedicationDispense2MedicationDispense(
 		// ImmunizationMedicationInformation) -> medication
 		if (cdaMedicationDispense.getProduct() != null && !cdaMedicationDispense.getProduct().isSetNullFlavor()) {
 			EntryResult fhirMedicationResult = tManufacturedProduct2Medication(cdaMedicationDispense.getProduct(),
-					bundleInfo);
-			if (fhirMedicationResult.hasResult()) {
-				result.updateFrom(fhirMedicationResult);
-				for (BundleEntryComponent entry : fhirMedicationResult.getBundle().getEntry()) {
-					if (entry.getResource() instanceof org.hl7.fhir.dstu3.model.Medication) {
-						Medication medicationResource = (Medication) entry.getResource();
-						fhirMediDisp.setMedication(getReference(medicationResource));
+					localBundleInfo);
+			result.updateFrom(fhirMedicationResult);
+			localBundleInfo.updateFrom(fhirMedicationResult);
+			for (BundleEntryComponent entry : fhirMedicationResult.getFullBundle().getEntry()) {
```

Comment:
FHIRUtil has a function for this findResource

Reply:
Gotchya, I'll add that